impeller/aiks/canvas.cc

```diff
@@ -681,6 +681,14 @@ void Canvas::SaveLayer(const Paint& paint,
   TRACE_EVENT0("flutter", "Canvas::saveLayer");
   Save(true, paint.blend_mode, backdrop_filter);
 
+  // The DisplayList bounds/rtree doesn't account for filters applied to parent
+  // layers, and so sub-DisplayLists are getting culled as if no filters are
+  // applied.
+  // See also: https://github.com/flutter/flutter/issues/139294
+  if (paint.image_filter) {
+    transform_stack_.back().cull_rect = std::nullopt;
```

Comment:
Another technique could be to run the current cull_rect through the filter to get a reverse bounds, as in:

https://github.com/flutter/engine/blob/c48b08ac46828542891a52f402072a6e76f04217/impeller/entity/contents/filters/filter_contents.h#L137

Reply:
Another optimization would be to see if the filter modifies location at all. The ColorFilterImageFilter would not need to clear the cull_rect, for instance.

Reply:
I don't think that's quite enough. In order to produce a correct cull rect, wouldn't we need to pass each new clip's bounds through the chain of all parent filters? Also, it would need to be forward bounds, not reverse, since the cull rect needs to expand for kernel filters.

Am I missing something?

Reply:
Yes, you are missing these lines:

https://github.com/flutter/engine/blob/832238a3b493bed828e9e6696c5b56b65a9abcda/display_list/dl_builder.cc#L418C1-L435C1

That happens on each restore as we back out of recording the DL...

Reply:
It is done for each filter on each restore:

https://github.com/flutter/engine/blob/832238a3b493bed828e9e6696c5b56b65a9abcda/display_list/dl_builder.cc#L418C1-L435C1